constraint/pkg/client/constraint_client.go

```diff
@@ -16,24 +17,58 @@ type constraintClient struct {
 	// constraint is a copy of the original Constraint added to Client.
 	constraint *unstructured.Unstructured
 
+	// matchers are the per-target Matchers for this Constraint.
+	matchers map[string]constraints.Matcher
+
 	// enforcementAction is what should be done if the Constraint is violated or
 	// fails to run on a review.
+	// passed to constraintMatchResult.enforcementAction.
 	enforcementAction string
 
-	// matchers are the per-target Matchers for this Constraint.
-	matchers map[string]constraints.Matcher
+	// enforcementActionsForEP stores precompiled enforcement actions for each enforcement point.
+	enforcementActionsForEP map[string][]string
 }
 
 func (c *constraintClient) getConstraint() *unstructured.Unstructured {
 	return c.constraint.DeepCopy()
 }
 
-func (c *constraintClient) matches(target string, review interface{}) *constraintMatchResult {
+func (c *constraintClient) matches(target string, review interface{}, enforcementPoints ...string) *constraintMatchResult {
 	matcher, found := c.matchers[target]
 	if !found {
 		return nil
 	}
 
+	enforcementActions := make(map[string]bool)
+	if apiconstraints.IsEnforcementActionScoped(c.enforcementAction) {
+		for _, ep := range enforcementPoints {
+			var actions []string
+			if ep == apiconstraints.AllEnforcementPoints {
```

Comment:
`*` cannot be provided to `matches()`, calls to `matches()` must provide a specific enforcement point.

Remember, `*` only makes sense when being specified on the constraint.

Reply:
Agreed. sorry for missing this. FIxed it.